### PR TITLE
Allow remote signing infrastructure to assign operational IDs

### LINKED ISSUE #13500: Commissioner must allow assignment of fabric and node IDs by signing infrastructure
#### Problem

There is a lot of logic entwined within the current commissioner code that assumes and relies upon behavior where the commissioner assigns fabric and node IDs.  But in some ecosystems, signers must assign these.

 #### Proposed Solution

* Provide commissioner interfaces that allow signers to assign IDs
* Honor assigned IDs in operational discovery

Without a paradigm change, a remote signer assigning an ID for a fabric to which the commissioner does not belong would of course break.  This is only relevant though if the commissioner needs to establish CASE, which is not necessarily needed for a split commissioner / admin topology where a separate node establishes CASE and issues CommissioningComplete.

### PR DESCRIPTION
**Pull request by msandstedt**
#### Problem

The operational credentials delegate interface includes the SetFabricIdForNextNOCRequest and SetNodeIdForNextNOCRequest methods, which are supposed to provide non-binding operational ID hints to signers.

However, these actually are not optional because no matter what identifiers the signer encloses in the certificate chain, the commissioner assumes the signer has honored its suggestions and will not be able to locate or communicate with the commissioned node unless it carries the identifiers the commissioner has proposed.

#### Change overview

This commit fixes the issue by using the signer-provided certificate chain as source of truth, extracting compressed fabric ID and node ID from this.

This will not disrupt signers that honor the commissioner's hints, but also works for signers that assign their own IDs.  In the case of a combined commissioner / admin, the signer is free to assign node IDs of its choice.  In the case of a standalone commissioner implementation that does not establish CASE or invoke CommissioningComplete, the signer is free to select all of root cert, fabric ID and node ID.  In all cases, the resulting PeerId struct in the CommissioneeDeviceProxy will carry identifiers that match the certificate chain from the signer.

Fixes #13500

#### Testing

* Unit tests have been added for the new crypto helpers that are extracting operational ID from the cert chain.
* Commissioning with commisisoner-selected node IDs was manually tested - no regressions.
* Commissioning with signer-selected node IDs was manually tested - this now works and didn't before.
* Existing chip-tool integration commissioning tests pass.

### CODE REVIEW

**Comment by msandstedt** [comment]
CC @samadDotDev , @kianooshkarami , @pan-apple , @cecille , @tcarmelveilleux

**Review by wbschiller — APPROVED** [review]
Looks good.

**Review by tcarmelveilleux — APPROVED** [review]

File: src/credentials/CHIPCert.cpp [hunk 1]
```diff
@@ -867,6 +867,17 @@ CHIP_ERROR ExtractNodeIdFabricIdFromOpCert(const ChipCertificateData & opcert, N
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ExtractOperationalDiscoveryIdFromRootPubKeyOpCert(const Crypto::P256PublicKey & rootPublicKey, ByteSpan noc, PeerId & id)
```
[the remark this answers is not in the document]
Comment (msandstedt) [reply src/credentials/CHIPCert.cpp:870]: Will do.

[the remark this answers is not in the document]
Comment (msandstedt) [reply src/credentials/CHIPCert.cpp]: Done.

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: unit tested code that has been up for sufficient time for cross timezone review.

**Pull request merged**